src/screencast/pipewire_screencast.c

```diff
@@ -155,13 +139,9 @@ static void pwr_handle_stream_add_buffer(void *data, struct pw_buffer *buffer) {
 			return;
 		}
 
-		// mmap buffer, so we can use the data_ptr in on_process
-		d[0].data = mmap(NULL, d[0].maxsize, PROT_READ | PROT_WRITE, MAP_SHARED, d[0].fd, d[0].mapoffset);
-		if (d[0].data == MAP_FAILED) {
-			logprint(ERROR, "pipewire: unable to mmap memory");
-			cast->err = 1;
-			return;
-		}
+		// create wl_buffer
+		d[0].data = import_shm_buffer(cast, d[0].fd, cast->screencopy_frame.format,
```

Comment:
Hmmmm. `d[0].data` is supposed to be raw pixel data, not a `wl_buffer`.

Instead we should use `pw_buffer.user_data` I think?

Reply:
could you please link me the header? I can't find a user_data.

Reply:
In `src/pipewire/stream.h`:

```c
struct pw_buffer {
	struct spa_buffer *buffer;	/**< the spa buffer */
	void *user_data;		/**< user data attached to the buffer */
	uint64_t size;			/**< This field is set by the user and the sum of
					  *  all queued buffer is returned in the time info */
};
```

The docs are at https://docs.pipewire.org/structpw__buffer.html but are down atm.

Reply:
thanks!